lms/djangoapps/grades/tests/test_api.py

```diff
@@ -108,3 +112,126 @@ def test_override_subsection_grade(self, earned_graded):
+        super().tearDown()
+        PersistentSubsectionGradeOverride.objects.all().delete()  # clear out all previous overrides
+
+    def test_clear_user_course_grades(self):
+        api.override_subsection_grade(
+            self.user.id,
+            self.course.id,
+            self.subsection.location,
+            overrider=self.overriding_user,
+            earned_graded=0.0,
+            comment='Test Override Comment',
+        )
+        override_obj = api.get_subsection_grade_override(
+            self.user.id,
+            self.course.id,
+            self.subsection.location
+        )
+        course_grade = PersistentCourseGrade.read(self.user.id, self.course.id)
+        self.assertIsNotNone(course_grade)
+        self.assertIsNotNone(override_obj)
+
+        api.clear_user_course_grades(self.user.id, self.course.id)
+
+        with self.assertRaises(PersistentCourseGrade.DoesNotExist):
+            PersistentCourseGrade.read(self.user.id, self.course.id)
+
+        with self.assertRaises(PersistentSubsectionGrade.DoesNotExist):
+            api.get_subsection_grade_override(
+                self.user.id,
+                self.course.id,
+                self.subsection.location
+            )
+
+    def test_clear_wrong_user_course_grades(self):
+        wrong_user = UserFactory()
+        api.override_subsection_grade(
+            self.user.id,
+            self.course.id,
+            self.subsection.location,
+            overrider=self.overriding_user,
+            earned_graded=0.0,
+            comment='Test Override Comment',
+        )
+        override_obj = api.get_subsection_grade_override(
+            self.user.id,
+            self.course.id,
+            self.subsection.location
+        )
+        course_grade = PersistentCourseGrade.read(self.user.id, self.course.id)
+        self.assertIsNotNone(course_grade)
+        self.assertIsNotNone(override_obj)
+
+        api.clear_user_course_grades(wrong_user.id, self.course.id)
+
+        after_clear_override_obj = api.get_subsection_grade_override(
+            self.user.id,
+            self.course.id,
+            self.subsection.location
+        )
+        after_clear_course_grade = PersistentCourseGrade.read(self.user.id, self.course.id)
+        self.assertIsNotNone(after_clear_override_obj)
+        self.assertIsNotNone(after_clear_course_grade)
+
+    @patch('lms.djangoapps.grades.models_api._PersistentSubsectionGrade')
+    @patch('lms.djangoapps.grades.models_api._PersistentCourseGrade')
+    def test_assert_clear_grade_methods_called(self, mock_course_grade, mock_subsection_grade):
+        api.clear_user_course_grades(self.user.id, self.course.id)
+        mock_course_grade.delete_course_grade_for_learner.assert_called_with(self.course.id, self.user.id)
+        mock_subsection_grade.delete_subsection_grades_for_learner.assert_called_with(self.user.id, self.course.id)
+
+    @patch('lms.djangoapps.grades.models_api._PersistentSubsectionGrade')
```

Comment:
I suggested in another comment that you remove the error handling in that function, so this test can be removed as well.